ui/src/app/utils/search/filter-nodes.ts

```diff
@@ -0,0 +1,177 @@
+import FilterHandlers from "./filter-handlers";
+import type { Filters, FilterValue, PrefixedFilter } from "./filter-handlers";
+
+export type GetValue<N> = (
+  node: N,
+  filter: string
+) => FilterValue | FilterValue[] | null;
+
+export default class FilterNodes<N, PK extends keyof N> extends FilterHandlers {
```

Comment:
Since the plan is for this to be used with discoveries as well, I think `FilterNodes` might be a bit of a misnomer since nodes in MAAS-land are specifically `Machine | Device | Controller`. Maybe `FilterObjects` if that's not too vague?

Reply:
For the generic utils we use `item` and `I` so maybe we should carry that over here (also on the state objects they're stored in an `items` array). Does that sound OK?

Reply:
Yeah that sounds good